### PR TITLE
IGAPP-778: Use airbnb config

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

This PR removes rules from our .eslintrc that are already part of the airbnb/standard config. Some rules appeared too strict for our project (for now at least). I will create one more PR with the two react rules set to off.

If you'd like any other rule that is currently disabled, let me know.

### CODE REVIEW

File: native/src/routes/Pois.tsx [hunk 1]
```diff
@@ -96,7 +96,9 @@ const Pois = ({
         )
         setSelectedFeature(currentFeature ?? null)
       }
-      location && setFeatureLocations(featureLocations)
+      if (location) {
+        setFeatureLocations(featureLocations)
+      }
```
Comment (klinzo) [inline native/src/routes/Pois.tsx:101]: @f1sh1918 @Taggotty is this if correct/necessary?

File: .eslintrc.js [hunk 2]
```diff
@@ -21,33 +23,52 @@ module.exports = {
...
+    'class-methods-use-this': 'off',
+    'default-case': 'off',
+    'global-require': 'off',
+    'import/extensions': 'off',
+    'import/no-extraneous-dependencies': 'off',
+    'import/no-unresolved': 'off',
+    'import/prefer-default-export': 'off',
+    'lines-between-class-members': 'off',
+    'no-shadow': 'off',
+    'no-nested-ternary': 'off',
```
Comment (SteffiStuffel) [inline .eslintrc.js:40]: Noooooo :sob: 

Seriously, I would think this could really make sense to have as a rule.

File: .eslintrc.js [hunk 3]
```diff
@@ -21,33 +23,52 @@ module.exports = {
...
-    '**/lib-dist/'
+    '**/lib-dist/',
+    '.eslintrc.js'
   ],
   rules: {
-    'arrow-body-style': 'error',
-    'arrow-parens': ['error', 'as-needed'],
+    // Overly strict rules (for now)
+    'class-methods-use-this': 'off',
+    'default-case': 'off',
```
Comment (SteffiStuffel) [inline .eslintrc.js:32]: Maybe this is also really helpful, as having no default can lead to unexpected results, when not thinking about the default case. Having this rule enforces to think about it.

File: .eslintrc.js [hunk 4]
```diff
@@ -21,33 +23,52 @@ module.exports = {
...
+    'global-require': 'off',
+    'import/extensions': 'off',
+    'import/no-extraneous-dependencies': 'off',
+    'import/no-unresolved': 'off',
+    'import/prefer-default-export': 'off',
+    'lines-between-class-members': 'off',
+    'no-shadow': 'off',
+    'no-nested-ternary': 'off',
+    'no-underscore-dangle': 'off',
+    'no-use-before-define': 'off',
```
Comment (SteffiStuffel) [inline .eslintrc.js:42]: Why not?

File: native/.idea/runConfigurations.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (SteffiStuffel) [inline native/.idea/runConfigurations.xml:1]: Why these changes?

File: native/jest.setup.ts [hunk 6]
```diff
@@ -25,7 +25,11 @@ const walkDir = (dir: string, callback: (filePath: string) => void): void => {
   fs.readdirSync(dir).forEach(f => {
     const filePath = path.join(dir, f)
     const isDirectory = fs.statSync(filePath).isDirectory()
-    isDirectory ? walkDir(filePath, callback) : callback(filePath)
+    if (isDirectory) {
```
Comment (SteffiStuffel) [inline native/jest.setup.ts:28]: Why not ternery?

File: native/src/components/NavigationTiles.tsx [hunk 7]
```diff
@@ -62,7 +62,7 @@ class NavigationTiles extends React.PureComponent<PropsType, StateType> {
     const contentSizeDiff = nativeEvent.contentSize.width - nativeEvent.layoutMeasurement.width
     this.setState({
       xPosition: nativeEvent.contentOffset.x,
-      contentSizeDiff: parseInt(contentSizeDiff.toFixed(0))
+      contentSizeDiff: parseInt(contentSizeDiff.toFixed(0), 10)
```
Comment (SteffiStuffel) [inline native/src/components/NavigationTiles.tsx:65]: Why is radix needed? It is optional.

**Review by SteffiStuffel — APPROVED** [review]

File: native/src/components/NavigationTiles.tsx (hunk 7 above)
Comment (klinzo) [reply native/src/components/NavigationTiles.tsx:65]: Yes it is, but from the docs:
> An integer between 2 and 36 that represents the radix (the base in mathematical numeral systems) of the string. Be careful—this does not default to 10! If the radix value is not of the Number type it will be coerced to a Number.

We could also disable the rule oc but I think it makes sense (or does not hurt at least). The corresponding rule is https://eslint.org/docs/rules/radix
> By default, parseInt() will autodetect decimal and hexadecimal (via 0x prefix). Prior to ECMAScript 5, parseInt() also autodetected octal literals, which caused problems because many developers assumed a leading 0 would be ignored.

This confusion led to the suggestion that you always use the radix parameter to parseInt() to eliminate unintended consequences. So instead of doing this:

File: native/jest.setup.ts (hunk 6 above)
Comment (klinzo) [reply native/jest.setup.ts:28]: https://eslint.org/docs/rules/no-unused-expressions
> An unused expression which has no effect on the state of the program indicates a logic error.

> For example, n + 1; is not a syntax error, but it might be a typing mistake where a programmer meant an assignment statement n += 1; instead. Sometimes, such unused expressions may be eliminated by some build tools in production environment, which possibly breaks application logic.

It is possible to disable this for ternaries though.

File: native/.idea/runConfigurations.xml (hunk 5 above)
Comment (klinzo) [reply native/.idea/runConfigurations.xml:1]: No idea, IntelliJ was quite persistent in re adding those even though I deleted them quite a few times :D

File: .eslintrc.js (hunk 3 above)
Comment (klinzo) [reply .eslintrc.js:32]: We already have `'@typescript-eslint/no-use-before-define': 'error'` which does the same with typescript support. This rule also throws the error all the time:
> 'React' was used before it was defined  no-use-before-define

File: .eslintrc.js (hunk 4 above)
Comment (klinzo) [reply .eslintrc.js:42]: We already have `'@typescript-eslint/no-use-before-define': 'error'` which does the same with typescript support. This rule also throws the error all the time:
> 'React' was used before it was defined  no-use-before-define

File: .eslintrc.js (hunk 3 above)
Comment (klinzo) [reply .eslintrc.js:32]: Will add https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/switch-exhaustiveness-check.md which only enforces necessary switch statements. I don't think it makes sense being forced to add unnecessary defaults that are not possible to be executed.

File: .eslintrc.js (hunk 2 above)
Comment (klinzo) [reply .eslintrc.js:40]: Added to TODOs, will do in a separate PR. If you want you can do it as well :)

File: native/src/components/NavigationTiles.tsx (hunk 7 above)
Comment (SteffiStuffel) [reply native/src/components/NavigationTiles.tsx:65]: yeah, oke, makes sense.

File: .eslintrc.js (hunk 2 above)
Comment (klinzo) [reply .eslintrc.js]: Done in #447

File: native/src/routes/Pois.tsx (hunk 1 above)
Comment (Taggotty) [reply native/src/routes/Pois.tsx:101]: Not sure. @f1sh1918 ?

File: web/src/components/HeaderActionItemDropDown.tsx [hunk 8]
```diff
@@ -89,8 +89,8 @@ const HeaderActionItemDropDown = (props: PropsType): ReactElement => {
 
   return (
     <Container ref={wrapperRef}>
-      <Tooltip text={text} flow={'down'} mediumViewportFlow={'left'}>
-        <button aria-label={text} onClick={toggleDropDown}>
+      <Tooltip text={text} flow='down' mediumViewportFlow='left'>
+        <button type='button' aria-label={text} onClick={toggleDropDown}>
```
Comment (Taggotty) [inline web/src/components/HeaderActionItemDropDown.tsx:93]: why did you add `type='button'`?

**Review by Taggotty — APPROVED** [review]
